build/buildall

```diff
@@ -161,6 +161,17 @@ if [[ "$DIST_PROFILE" == *Scala213 ]]; then
   SCALA213=1
 fi
 
+if [[ $SCALA213 == 0 && ${DIST_PROFILE[@]} =~ "400" ]]; then 
+  echo "Spark 4.0.0 is only supported for Scala 2.13, please change the Scala version or remove 4.0.0 from the dist profile"
+  exit 1 
+fi
+
+# Spark 4.0.0 is only compiled with Java 17 
+JAVA_MAJOR_VER=$(java -version 2>&1 | awk -F[\"_] 'NR==1{print $2}' | awk -F. '{print $1}')
+if [[ ${DIST_PROFILE[@]} =~ "400" && $JAVA_MAJOR_VER -lt "17" ]]; then 
+  echo "Spark 4.0.0 support is only available for JDK 17+"
+  exit 1 
+fi
```

Comment:
This check should be delegated to Maven enforce plugin similar to https://github.com/NVIDIA/spark-rapids/blob/c7129f5a006e19e4dc592dfd3810c1ac99f7f263/pom.xml#L679-L682

Reply:
We only want to enforce Java 17 when we are building for Spark 4.0.0. but `requireJavaVersion` doesn't take in a condition AFAIK.

Reply:
obviously we need to put spark400-specific requirements in the spark400 profile

Reply:
Thanks for the pointer. I have removed the change from build/buildall script and delegated the Java 17 check, just like the check to disallow 400 build with Scala 2.12, to maven enforcer plugin